pkg/manager/internal.go

```diff
@@ -584,3 +613,13 @@ func (cm *controllerManager) startLeaderElection() (err error) {
 func (cm *controllerManager) Elected() <-chan struct{} {
 	return cm.elected
 }
+
+func (cm *controllerManager) startRunnable(r Runnable) {
+	cm.waitForRunnable.Add(1)
+	go func() {
+		defer cm.waitForRunnable.Done()
+		if err := r.Start(cm.internalStop); err != nil {
+			cm.errChan <- err
```

Comment:
aren't we gonna have the same issue we had before, where if multiple things error out at the same time, we'll have some block forever since the error channel isn't drained?  That's why we had the weird error-signaler construct before

Reply:
I don't think so because we do read from it in `engageStopProcedure` (but only log them rather than returning).

I have added a comment clarifying everything that writes into the channel must not block the `Start` func to make sure we get to `engageStopProcedure` and changed the latter to do the error reading before acquiring the lock to avoid deadlocks.

Reply:
ah, yeah, missed the "for" in `engageStopProceedure`